breakpad-symbols/src/http.rs

```diff
@@ -181,6 +182,73 @@ fn commit_cache_file(mut temp: NamedTempFile, final_path: &Path, url: &Url) -> i
     Ok(())
 }
 
+/// The result of a lookup by code_file/code_identifier against a symbol
+/// server.
+#[derive(Debug, Clone)]
+pub struct DebugInfoResult {
+    debug_file: String,
+    debug_identifier: DebugId,
+}
+
+/// Given a vector of symbol urls and a module with a code_file and code_identifier,
+/// this tries to request a symbol file using the code file and code identifier.
+///
+/// `<code file>/<code identifier>/<code file>.sym`
+///
+/// If the symbol server returns an HTTP 302 redirect, the Location header will
+/// have the correct download API url with the debug file and debug identifier.
+///
+/// This is supported by tecken
+///
+/// This returns a DebugInfoResult with the new debug file and debug identifier
+/// or None.
+async fn lookup_debug_info_by_code_info(
+    symbol_urls: &Vec<Url>,
+    module: &(dyn Module + Sync),
+) -> Option<DebugInfoResult> {
+    let lookup_path = match code_info_breakpad_sym_lookup(module) {
+        Some(value) => value,
+        _ => return None,
+    };
+
+    for base_url in symbol_urls {
+        let url = base_url.join(&lookup_path).ok()?;
+
+        debug!("Trying code file / code identifier lookup: {}", url);
+
+        // This should not follow redirects--we want the next url if there is one
+        let no_redirects_client = Client::builder()
+            .redirect(redirect::Policy::none())
+            .build()
+            .unwrap();
+
+        let response = no_redirects_client.get(url.clone()).send().await;
```

Comment:
I'm on the fence about this API. On the one hand, it's pretty convenient for other use cases to support requesting by code file and code id using the download API and being able to follow redirects to the final file in S3. On the other hand, it's pretty wonky for this case. Maybe it'd be better to have a new lookup API where you give it a code file and code identifier by querystring params and it returns a debug file and debug identifier in a JSON payload.

We could support multiple ways to get the information.

Are there other things we might use this functionality for that would inform what we should do?